ir/function.cpp

```diff
@@ -48,6 +47,40 @@ JumpInstr::it_helper BasicBlock::targets() const {
   return {};
 }
 
+void BasicBlock::replaceTargetWith(const BasicBlock *from,
```

Comment:
this code should patch instructions rather than deleting them & creating new ones

Reply:
The target basicblocks are stored as reference in those branch/switch instructions, should I change it to pointers?

Reply:
Hi Nuno, I've pushed the commit which makes the branch and switch to store the pointer rather than references to support patching, which to me is a little nasty. Feel free to revert it.